crates/loader/src/lib.rs

```diff
@@ -40,6 +40,14 @@ pub async fn from_file(
     loader.load_file(path).await
 }
 
+/// Load a Spin locked app from a standalone Wasm file.
+pub async fn from_wasm_file(wasm_path: impl AsRef<Path>) -> Result<LockedApp> {
+    let app_root = std::env::current_dir()?;
```

Comment:
Actually, I believe you already have `working_dir` where this function is called; could probably just pass that through.

Reply:
This is the directory relative to which to resolve the Wasm file reference (and, perhaps, in future, other paths).  If we use `working_dir` then we have to copy the Wasm file first (before copying it again), and any file-related errors will show unexpected locations.

The alternative is for `app_root` to be the directory containing the Wasm file, and to override `wasm_path` to be only the name of the Wasm file (so that the relative reference resolves correctly), but I'm not sure that's going to be helpful with hypothetical future features where the user will expect paths entered on the CLI to be resolved relative to the current directory.  Of course it's hard to know because we haven't defined those features yet!

We could also do a more invasive change that threads multiple "app roots" through the loader, or redefines the app root, or absolutises everything earlier.  Or we can have a separate loader path for much longer than this PR does it, e.g. doing any copies needed then jumping straight to building the lockfile instead of leaning on the existing manifestful path.

Reply:
Sounds complicated. I retract my feelings.